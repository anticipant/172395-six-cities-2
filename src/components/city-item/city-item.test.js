import React from 'react';
import renderer from 'react-test-renderer';

import CityItem from "./city-item";

describe(`how city items render`, () => {

  it(`CityItem correctly renders after relaunch`, () => {
    const cityItem = {id: 111, name: `Moscow 20222`};
    const tree = renderer.create(<CityItem
      item = {cityItem}
      onItemClick = {() => {}}
      activeCity={`Minsk`}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it(`CityItem correctly renders with active city`, () => {
    const cityItem = {id: 564, name: `london`};
    const tree = renderer.create(<CityItem
      item = {cityItem}
      onItemClick = {() => {}}
      activeCity={`london`}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
