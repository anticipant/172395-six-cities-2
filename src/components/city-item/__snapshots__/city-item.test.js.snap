// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`how city items render CityItem correctly renders after relaunch 1`] = `
<li
  className="locations__item"
  onClick={[Function]}
>
  <a
    className="locations__item-link tabs__item "
    href="#"
  >
    <span>
      Moscow 20222
    </span>
  </a>
</li>
`;

exports[`how city items render CityItem correctly renders with active city 1`] = `
<li
  className="locations__item"
  onClick={[Function]}
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    href="#"
  >
    <span>
      london
    </span>
  </a>
</li>
`;
